src/internal/connector/exchange/handlers.go

```diff
@@ -0,0 +1,88 @@
+package exchange
+
+import (
+	"context"
+
+	"github.com/alcionai/corso/src/internal/connector/graph"
+	"github.com/alcionai/corso/src/pkg/backup/details"
+	"github.com/alcionai/corso/src/pkg/fault"
+	"github.com/alcionai/corso/src/pkg/path"
+	"github.com/alcionai/corso/src/pkg/services/m365/api"
+)
+
+// ---------------------------------------------------------------------------
+// restore
+// ---------------------------------------------------------------------------
+
+type restoreHandler interface {
+	itemRestorer
+	containerAPI
+	newContainerCache(userID string) graph.ContainerResolver
+	formatRestoreDestination(
+		destinationContainerName string,
+		collectionFullPath path.Path,
+	) *path.Builder
+}
+
+// runs the item restoration (ie: item creation) process
+// for a single item, whose summary contents are held in
+// the body property.
+type itemRestorer interface {
+	restore(
+		ctx context.Context,
+		body []byte,
+		userID, destinationID string,
+		errs *fault.Bus,
+	) (*details.ExchangeInfo, error)
+}
+
+// runs the actual graph API post request.
+type itemPoster[T any] interface {
+	PostItem(
+		ctx context.Context,
+		userID, dirID string,
+		body T,
+	) (T, error)
+}
+
+// produces structs that interface with the graph/cache_container
+// CachedContainer interface.
+type containerAPI interface {
+	// POSTs the creation of a new container
+	CreateContainer(
+		ctx context.Context,
+		userID, containerName, parentContainerID string,
+	) (graph.Container, error)
+
+	// GETs a container by name
```

Comment:
Ah, good call, I should have commented that.  It's supposed to represent whether the feature is supported by the interface.  But like you pointed out in an earlier comment, checking for a nil response is just as good here.

And, actually, I'm going to go one step further.  Instead of returning another interface, I'm going to return a bool directly.  Suppliers that support the func will call their internal client.  Those that don't will automatically return false.